src/Support/HigherOrderMessageCollection.php

```diff
@@ -29,8 +29,14 @@ public function add(string $filename, int $line, string $methodName, array $argu
      */
     public function chain(object $target): void
     {
+        $originalTarget = $target;
+
         foreach ($this->messages as $message) {
             $target = $message->call($target);
+
+            if ($target === null) {
+                $target = $originalTarget;
```

Comment:
Maybe you can change this to use the last non null target?

Reply:
sure